src/components/Timeline/TimelineNote.tsx

```diff
@@ -1,13 +1,73 @@
 import { GiftCardDetailsQuery, OrderEventFragment } from "@dashboard/graphql";
-import { getUserInitials, getUserName } from "@dashboard/misc";
-import { Box, Button, EditIcon, Text } from "@saleor/macaw-ui-next";
+import { getUserName } from "@dashboard/misc";
+import { staffMemberDetailsUrl } from "@dashboard/staff/urls";
+import { Box, Button, EditIcon, Text, vars } from "@saleor/macaw-ui-next";
+import { InfoIcon, LinkIcon, MessageSquareIcon } from "lucide-react";
 import { useState } from "react";
 import { FormattedMessage } from "react-intl";
+import { Link } from "react-router-dom";
 
-import { DashboardCard } from "../Card";
 import { DateTime } from "../Date";
-import { UserAvatar } from "../UserAvatar";
 import { TimelineNoteEdit } from "./TimelineNoteEdit";
+import { safeStringify } from "./utils";
+
+// CSS for hover effect
+const noteCardStyles = `
+  .timeline-note-card:hover {
+    background-color: ${vars.colors.background.default2};
+  }
+  .timeline-note-row .timeline-info-icon {
+    opacity: 0;
+    transition: opacity 0.15s ease-in-out;
+  }
+  .timeline-note-row:hover .timeline-info-icon {
+    opacity: 1;
+  }
+  .timeline-note-card .timeline-edit-button {
+    opacity: 0;
+    transition: opacity 0.15s ease-in-out;
+  }
+  .timeline-note-card:hover .timeline-edit-button {
+    opacity: 1;
+  }
+  .timeline-edit-link,
+  .timeline-user-link {
+    text-decoration: none;
+  }
+  .timeline-edit-link:hover,
+  .timeline-user-link:hover,
+  .timeline-edit-link:hover span,
+  .timeline-user-link:hover span {
+    text-decoration: underline;
+  }
+`;
```

Comment:
issue: let's use CSS Modules if we need to have custom styles 🙏🏻 - we already have 3 different ways of customizing styles and CSS Modules are by far the most used and cleanest